test/test_common/utility.cc

```diff
@@ -187,30 +187,27 @@ std::vector<std::string> TestUtility::split(const std::string& source, const std
 }
 
 void TestUtility::renameFile(const std::string& old_name, const std::string& new_name) {
-#if !defined(WIN32)
-  const int rc = ::rename(old_name.c_str(), new_name.c_str());
-  ASSERT_EQ(0, rc);
-#else
+#if defined(WIN32)
   // use MoveFileEx, since ::rename will not overwrite an existing file. See
   // https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/rename-wrename?view=vs-2017
   const BOOL rc = ::MoveFileEx(old_name.c_str(), new_name.c_str(), MOVEFILE_REPLACE_EXISTING);
   ASSERT_NE(0, rc);
+#else
+  const int rc = ::rename(old_name.c_str(), new_name.c_str());
+  ASSERT_EQ(0, rc);
 #endif
 };
 
 void TestUtility::createDirectory(const std::string& name) {
-#if !defined(WIN32)
-  ::mkdir(name.c_str(), S_IRWXU);
-#else
+#if defined(WIN32)
   ::_mkdir(name.c_str());
+#else
+  ::mkdir(name.c_str(), S_IRWXU);
 #endif
 }
 
 void TestUtility::createSymlink(const std::string& target, const std::string& link) {
-#if !defined(WIN32)
-  const int rc = ::symlink(target.c_str(), link.c_str());
-  ASSERT_EQ(rc, 0);
-#else
+#if defined(WIN32)
```

Comment:
I've seen you use `_MSC_VER` elsewhere and WIN32 here. Why's that? If possible can we use just one condition?

Also why `#if defined(WIN32)` rather than the more concise `#ifdef WIN32`?

Usually I see the `define(xxx)` form in the context of a more complex logical expression: `if defined(WIN32) || defined(APPLE)`